src/main/java/seedu/module/logic/commands/DeleteTagCommand.java

```diff
@@ -63,7 +63,7 @@ public CommandResult execute(Model model) throws CommandException {
                 taskToTag.getModule(), taskToTag.getDescription(), taskToTag.getWorkload(),
                 taskToTag.getDoneStatus(), newTags);
 
-        if (!taskToTag.isSameTask(editedTask) && model.hasTask(editedTask)) {
+        if (oldTags.equals(newTags)) {
```

Comment:
I recommend moving this block of code to before editedTask. Avoids unnecessary generation of the editedTask object if there's no tag to delete.

Reply:
agreed

Reply:
resolved :)